buildpacks/maven/src/main.rs

```diff
@@ -201,30 +209,51 @@ impl Buildpack for MavenBuildpack {
         // running since they might be confusing to the user.
         let internal_maven_options = vec![String::from("-B")];
 
-        log_header("Executing Maven");
-        log_info(format!(
-            "$ {} {} {}",
-            mvn_executable.to_string_lossy(),
-            shell_words::join(&maven_options),
-            shell_words::join(&maven_goals)
-        ));
-
-        util::run_command(
-            Command::new(&mvn_executable)
+        output::print_section("Running Maven build");
+        output::print_subsection(BuildpackOutputText::new(vec![
+            BuildpackOutputTextSection::regular("Running "),
+            BuildpackOutputTextSection::value(format!(
+                "{} {} {}",
+                mvn_executable.to_string_lossy(),
+                shell_words::join(&maven_options),
+                shell_words::join(&maven_goals)
+            )),
+        ]));
+
+        output::track_timing(|| {
+            let mut command = Command::new(&mvn_executable);
+
+            command
                 .current_dir(&context.app_dir)
                 .args(
                     maven_options
                         .iter()
                         .chain(&internal_maven_options)
                         .chain(&maven_goals),
                 )
-                .envs(&mvn_env),
-            MavenBuildpackError::MavenBuildIoError,
-            MavenBuildpackError::MavenBuildUnexpectedExitCode,
-        )?;
-
-        util::run_command(
-            Command::new(&mvn_executable)
+                .envs(&mvn_env);
+
+            output::run_command(
+                command,
+                false,
+                MavenBuildpackError::MavenBuildIoError,
+                |output| MavenBuildpackError::MavenBuildUnexpectedExitCode(output.status),
+            )
+        })?;
```

Comment:
- There's a partial command emitted earlier but on failure we don't have the full command that was run, needed for debugging. Please add it or emit the entire command.

Reply:
I moved this feedback to a "lster" issue https://github.com/heroku/buildpacks-jvm/issues/762